libs/core-ui/src/lib/util/ObjectDetectionStatisticsUtils.ts

```diff
@@ -15,13 +15,26 @@ export enum ObjectDetectionMetrics {
 }
 
 export const generateObjectDetectionStats: (
-  selectionIndexes: number[][]
+  selectionIndexes: number[][],
+  objectDetectionCache: Map<string, [number, number, number]>,
+  objectDetectionInputs: [string, string, number]
 ) => ILabeledStatistic[][] = (
-  selectionIndexes: number[][]
+  selectionIndexes: number[][],
+  objectDetectionCache: Map<string, [number, number, number]>,
+  objectDetectionInputs: [string, string, number]
 ): ILabeledStatistic[][] => {
   return selectionIndexes.map((selectionArray) => {
     const count = selectionArray.length;
 
+    const key: [number[], string, string, number] = [
+      selectionArray,
+      objectDetectionInputs[0],
```

Comment:
Should it be a well-defined object (class instance) so that it's clear which is which? Right now, it's possible to return something in wrong order from the SDK without noticing here.

Reply:
@romanlutz Currently all the inputs are stitched and serialized to a string key of the format "<cohort indices>,agg method, class name, iou threshold"

If I define a new class with those 4 stored in instance variables, I may still end up manually specifying the order like `[object.cohort_indices, object.agg_method,...].toString()`. Perhaps this'll be more resilient that passing in the order of variables multiple times in places of typed variables.

Let me know your suggestion, thanks a lot!

Reply:
It comes from the SDK, right? So you can use a well-defined object there, too 🙂 or I'm misunderstanding where this order comes from

Reply:
Yeah I must have been thinking of a different PR.... this is a cache lookup so it's local enough that I'm fine with this.

Reply:
Ahh I see, thanks for reviewing this!